internal/configload/configload.go

```diff
@@ -0,0 +1,279 @@
+		Pass:    ic.Tests{},
+		Fail:    ic.Tests{},
+		Skip:    ic.Tests{},
+		Ignore:  ic.Tests{},
+	}
+
+	//nolint:govet // we don't care about alignment there
+	for _, testCategory := range []struct { // testCategory examples: pass, skip sections in the yaml file
+		yamlTests []any     // taken from the file, yaml representation of tests, incoming tests
+		outTests  *ic.Tests // yamlTests transformed to the internal representation
+	}{
+		{tc.Pass, &t.Pass},
+		{tc.Fail, &t.Fail},
+		{tc.Skip, &t.Skip},
+		{tc.Ignore, &t.Ignore},
+	} {
+		for _, test := range testCategory.yamlTests {
+			switch test := test.(type) {
+			case map[string]any:
+				keys := maps.Keys(test)
+				if len(keys) != 1 {
+					return nil, fmt.Errorf("invalid syntax: expected 1 element, got: %v", keys)
+				}
+
+				var arrPointer *[]string
+
+				k := keys[0]
+				switch k {
+				case "regex":
+					arrPointer = &testCategory.outTests.NameRegexPattern
+				case "not_regex":
+					arrPointer = &testCategory.outTests.NameNotRegexPattern
+				case "output_regex":
+					arrPointer = &testCategory.outTests.OutputRegexPattern
+				default:
+					return nil, fmt.Errorf("invalid field name %q", k)
+				}
+
+				mValue := test[k]
+
+				regexp, ok := mValue.(string)
+				if !ok {
+					// Arrays are illegal:
+					// - regex:
+					//   - foo
+					//   - bar
+					if _, ok := mValue.([]string); ok {
+						return nil, fmt.Errorf("invalid syntax: %s value shouldn't be an array", k)
+					}
+
+					return nil, fmt.Errorf("invalid syntax: expected string, got: %T", mValue)
+				}
+
+				// i.e. pointer to testCategory.outTests.RegexPattern = append(testCategory.outTests.RegexPattern, regexp)
+				*arrPointer = append(*arrPointer, regexp)
+
+				continue
+
+			case string:
+				testCategory.outTests.Names = append(testCategory.outTests.Names, test)
+				continue
+
+			default:
+				return nil, fmt.Errorf("invalid type of %[1]q: %[1]T", test)
+			}
+		}
+	}
+
+	return &t, nil
+}
+
+// fillAndValidate populates the configuration with default values and performs validation.
+func (c *config) fillAndValidate() error {
+	commonDefault := &c.Results.Common.Default
+
+	validStatus := func(status ic.Status) bool {
+		s := ic.Status(strings.ToLower(string(status)))
+		_, ok := ic.KnownStatuses[s]
+
+		return ok
+	}
+
+	if *commonDefault == "" {
+		*commonDefault = ic.Pass
+	}
+
+	if !validStatus(*commonDefault) {
+		return fmt.Errorf("invalid default result: %q", *commonDefault)
+	}
+
+	for _, r := range []*testConfig{
+		c.Results.FerretDB,
+		c.Results.MongoDB,
+	} {
+		if r == nil {
+			continue
+		}
+		origDefault := &r.Default
+
+		if *origDefault == "" {
+			continue
+		}
+
+		if !validStatus(r.Default) {
+			return fmt.Errorf("invalid default result: %q", *origDefault)
+		}
+
+		if *commonDefault != "" && r.Default != "" {
+			return errors.New("default value cannot be set in common, when it's set in database")
+		}
+
+		// XXX this doesn't work
+		r.Default = *commonDefault
```

Comment:
We can't set the default value here, so we have to set it [here](https://github.com/FerretDB/dance/pull/570/files#diff-54c7c1af5fa8d5db4dc49f0e8e80e93ba2b1183ba4d5c9e2e5729e6deae6a3cdR181) instead which makes this method incomplete.